test-harness/index.ts

```diff
@@ -26,17 +26,21 @@ describe('cli acceptance tests', () => {
 
     beforeAll(async () => {
       await Promise.all(
-        scenario.assets.map(async ([asset, contents]) => {
+        scenario.assets.map(async ([asset]) => {
           const tmpFileHandle = await tmpFile();
           tmpFileHandles.set(asset, tmpFileHandle);
 
-          replacements[asset] = tmpFileHandle.name;
-          replacements[`${asset}|no-ext`] = tmpFileHandle.name.replace(
-            new RegExp(`${path.extname(tmpFileHandle.name)}$`),
-            '',
-          );
+          const normalizedName = normalize(tmpFileHandle.name);
+
+          replacements[asset] = normalizedName;
+          replacements[`${asset}|no-ext`] = normalizedName.replace(new RegExp(`${path.extname(normalizedName)}$`), '');
+        }),
+      );
 
-          await writeFileAsync(tmpFileHandle.name, contents, { encoding: 'utf8' }); // todo: apply replacements to contents
+      await Promise.all(
```

Comment:
@P0lip Splitting this in two parts protects us from race conditions (eg. before replacing the name of the document in the ruleset, we need to be sure the document tempory storage file has been created and its location has been inserted in the replacements dictionary)

Reply:
Yeah, sounds good to me.